sentry_sdk/scope.py

```diff
@@ -635,12 +853,14 @@ def add_breadcrumb(self, crumb=None, hint=None, **kwargs):
         :param hint: An optional value that can be used by `before_breadcrumb`
             to customize the breadcrumbs that are emitted.
         """
-        client = kwargs.pop("client", None)
-        if client is None:
+        client = Scope.get_client()
+
+        if not client.is_active():
+            logger.info("Dropped breadcrumb because no client bound")
             return
 
-        before_breadcrumb = client.options.get("before_breadcrumb")
-        max_breadcrumbs = client.options.get("max_breadcrumbs")
+        before_breadcrumb = client.options["before_breadcrumb"]
```

Comment:
Why no longer the `.get`? Are we sure this is not `None`, or are we making sure to handle any errors correctly?

Reply:
Also below, for `max_breadcrumbs`

Reply:
changed